src/app/[lang]/account/page.tsx

```diff
@@ -0,0 +1,167 @@
+import Image from "@/components/image"
+import { BaseButton } from "@/components/material/base-button"
+import { Divider } from "@/components/material/divider"
+import { PageHeader } from "@/components/page-header"
+import { PageSelector } from "@/components/page-selector"
+import { Event, Semester } from "@/data/types"
+import { filterEventsAttendance, filterEventsAttendancePoints, getEvent } from "@/data/webData"
+import { getActiveSession } from "@/lib/oauth"
+import { Locale, getDictionary } from "@/localization"
+import { cookies } from "next/headers"
+import { redirect } from "next/navigation"
+import { FutureEventItem } from "../events/page"
+import { getMaximumSemesterDate, getMinimumSemesterDate, getSemester } from "@/lib/utils"
+import { InputSection } from "@/components/input/input-section"
+
+const entriesPerPage = 30
+
+interface AttendanceSemester {
+    date: Date
+    semester: Semester
+    items: Event[]
+}
+
+export default async function AccountPage(
+    {
+        params,
+        searchParams
+    }: {
+        params: {
+            lang: Locale
+        },
+        searchParams: {
+            page: string
+        }
+    }
+) {
+
+    // get the current session
+    const session = await getActiveSession(cookies())
+
+    // redirect to the login route if not logged in
+    if (session === null) return redirect("/api/oauth?refer=/account")
+
+    // get the lang dict
+    const langDict = await getDictionary(params.lang)
+
+    // get the user from the session
+    const user = session.user
+    const firstName = user.givenName
+    const email = user.email
+
+    // handle date/semester stuff
+    const dateNow = new Date()
+    const currentSemester = getSemester(dateNow)
+    const minSemesterDate = getMinimumSemesterDate(currentSemester)
+    const maxSemesterDate = getMaximumSemesterDate(currentSemester)
+
+    // get the user's attendance points for this semester
+    const attendancePointsResult = await filterEventsAttendancePoints({
+        userEmails: [email],
+        maxEntries: 1,
+        fromDate: minSemesterDate,
+        toDate: maxSemesterDate
+    })
+    const attendancePointsTotal = attendancePointsResult.results[0]?.points || 0
+
+    // parse search params
+    const currentPage = Math.max(Number.parseInt(searchParams.page) || 0, 0) // parse the page search parameter, ensuring that it is >= 1
```

Comment:
To ensure this result is >= 1, shouldn't this be or conditioned with 1 instead of 0? the max of 0 and 0 is 0.
```jsx
const currentPage = Math.max(Number.parseInt(searchParams.page) || 1, 0)
```

Reply:
The page value starts at 0; so the first page = 0, second page = 1, etc...